modal_proto/api.proto

```diff
@@ -1112,9 +1112,11 @@ message DictPopResponse {
 message DictUpdateRequest {
   string dict_id = 1 [ (modal.options.audit_target_attr) = true ];
   repeated DictEntry updates = 2;
+  bool if_not_exists = 3;
 }
 
 message DictUpdateResponse {
+  bool created = 1;
```

Comment:
You're not adding support for this in `.update` right now and I think that's fine, but this _may_ cause issues in the future if we want to do that, since presumably the updates would be optional at the level of individual keys rather than whole payloads.